### PR TITLE
222-pdf-show-amount-of-tax-on-invoice

### LINKED ISSUE #126: TEST: investigate flaky test
https://docs.codecov.com/docs/test-analytics

### PR DESCRIPTION
**Pull request by Segelzwerg**
Also closes #126

### CODE REVIEW

**Review by iTitus — COMMENTED** [review]
We might lose decimals when calculating the tax and the the totals
Can you make sure when adding up the net and the taxes even with only 2 digits after the comma it adds up?

**Comment by Segelzwerg** [comment]
> We might lose decimals when calculating the tax and the the totals Can you make sure when adding up the net and the taxes even with only 2 digits after the comma it adds up?

You are concerned that the net totals + VAT != total, because of rounding issues?

**Comment by iTitus** [comment]
> > We might lose decimals when calculating the tax and the the totals Can you make sure when adding up the net and the taxes even with only 2 digits after the comma it adds up?
> 
> Are you concerned that the net totals + VAT != total, because of rounding issues?

yes

**Comment by Segelzwerg** [comment]
> > > We might lose decimals when calculating the tax and the the totals Can you make sure when adding up the net and the taxes even with only 2 digits after the comma it adds up?
> > 
> > 
> > Are you concerned that the net totals + VAT != total, because of rounding issues?
> 
> yes

Let me build some tests and we see what happens.

**Comment by Segelzwerg** [comment]
> > > > We might lose decimals when calculating the tax and the the totals Can you make sure when adding up the net and the taxes even with only 2 digits after the comma it adds up?
> > > 
> > > 
> > > Are you concerned that the net totals + VAT != total, because of rounding issues?
> > 
> > 
> > yes
> 
> Let me build some tests and we see what happens.

@iTitus  I built a test. I think that should do it, right? The manual sum is a different operation than in the code.

File: invoice/tests.py [hunk 1]
```diff
@@ -598,6 +634,32 @@ def test_due_date_equal_date(self):
...
+        for item in invoice_items:
+            item.invoice = invoice
+            item.save()
+        net_total = sum(item.net_total for item in invoice_items)
+        net_total_string = f'{net_total:.2f} EUR'
+        vat_total = sum(item.tax_amount for item in invoice_items)
+        vat_total_string = f'{vat_total:.2f} EUR'
+        total_total = sum(item.total for item in invoice_items)
+        total_total_string = f'{total_total:.2f} EUR'
+        manual_total = net_total + vat_total
```
Comment (iTitus) [inline invoice/tests.py:652]: this is not rounded yet I think.
We want to check of the net total that is visible in the UI (two digits after comma) plus the tax amount that is visible in the UI (two digits after comma) equal the visible total (two digits after comma) exactly.
You are adding the exact values here and then rounding after.

**Review by iTitus — CHANGES_REQUESTED** [review]

File: invoice/tests.py (hunk 1 above)
Comment (Segelzwerg) [reply invoice/tests.py:652]: Isn't that the lines 653 & 661?

Comment (iTitus) [reply invoice/tests.py]: no, i updated the test

**Review by iTitus — APPROVED** [review]

**Comment by iTitus** [comment]
@Segelzwerg needs your explicit approval as well

**Pull request merged**